nomad/structs/structs.go

```diff
@@ -69,6 +69,9 @@ type QueryOptions struct {
 	// If set, any follower can service the request. Results
 	// may be arbitrarily stale.
 	AllowStale bool
+
+	// If set, used as prefix for resource list searches
+	Prefix string
```

Comment:
I created it as a string and not as a boolean, as that would require an additional parameter with the actual prefix string.